### ISSUE TITLE
Resolving JSON schema failing

### ISSUE THREAD

**Issue by simondean** [wework/speccy#36]
I'm seeing a bug when resolving a YAML spec that references a JSON schema.  The resolved JSON schema ends up embedded into the spec as a single stringified string value and not as a nested object.  It's affecting both the`resolve` and `lint` modes of Speccy.  

I've found the issue is due to a syntax error in the JSON schema file.  Speccy ignores the error and resolves the reference as a string instead of as a nested object.  

## Detailed description

I have `openapi.yaml` file like this: 

```yaml
openapi: 3.0.0
info: 
  title: Title
  description: Description
  version: 1.0.0
servers:
  - url: https://host
    description: Description
paths:
  /example:
    get: 
      description: Description
      operationId: getExample
      responses:
        '200':
          description: Example response
          content:
            application/json:
              schema:
                $ref: example-schema.json
```

The `example-schema.json` looks like the following.  I've created a syntax error in the JSON by removing the `,` between the type field and the properties field.  

```json
{
  "type": "object"
  "properties": {
    "example": {
      "type": "string"
    }
  }
}
```

I've ran the resolve mode with the following command: 

```bash
node_modules/.bin/speccy resolve openapi.yaml -o resolved.yaml
```

The `resolved.yaml` file outputted by Speccy looks like the following.  The schema has been embedded in a string rather than an error being raised by Speccy.  

```yaml
openapi: 3.0.0
info:
  title: Title
  description: Description
  version: 1.0.0
servers:
  - url: 'https://host'
    description: Description
paths:
  /example:
    get:
      description: Description
      operationId: getExample
      responses:
        '200':
          description: Example response
          content:
            application/json:
              schema: "{\n  \"type\": \"object\"\n  \"properties\": {\n    \"example\": {\n      \"type\": \"string\"\n    }\n  }\n}\n"
```

## Context

The lint mode was failing with a messaging saying it was not expecting the schema to be a string.  The error it printed out said it expected the schema to be an object and not a string

It was difficult to track down the root cause.  It would be much quick to diagnose for future people if Speccy raised an error for the root cause which was that referenced JSON file contained a syntax error.  

## Possible implementation

Raise an error when a referenced JSON or YAML file contains a syntax error.  

## Your environment

* Mac
* Node v9.10.0

**Comment by MikeRalphson** [comment]
Do you get a warning if you specify `--verbose`?

**Comment by simondean** [comment]
Just checked and yes there's a warning if I specify `--verbose`

Without verbose: 

```bash
$ node_modules/.bin/speccy lint openapi.yaml
Specification schema is invalid.

#/paths/~1example/get/responses/200/content/application~1json
expected '﻿{\n  "type": "object"\n  "properties": {\n    "example": {\n      "type": "string"\n    }\n  }\n}\n' to be an object
    expected '﻿{\n  "type": "object"\n  "properties": {\n    "example": {\n      "type": "string"\n    }\n  }\n}\n' to have type object
        expected 'string' to be 'object'
```

With verbose: 

```bash
$ node_modules/.bin/speccy lint openapi.yaml --verbose
GET openapi.yaml
GET example-schema.json 
{ YAMLException: missed comma between flow collection entries at line 3, column 3:
      "properties": {
      ^
    at generateError (node_modules/js-yaml/lib/js-yaml/loader.js:165:10)
    at throwError (node_modules/js-yaml/lib/js-yaml/loader.js:171:9)
    at readFlowCollection (node_modules/js-yaml/lib/js-yaml/loader.js:691:7)
    at composeNode (node_modules/js-yaml/lib/js-yaml/loader.js:1333:11)
    at readBlockMapping (node_modules/js-yaml/lib/js-yaml/loader.js:1009:16)
    at composeNode (node_modules/js-yaml/lib/js-yaml/loader.js:1332:12)
    at readDocument (node_modules/js-yaml/lib/js-yaml/loader.js:1492:3)
    at loadDocuments (node_modules/js-yaml/lib/js-yaml/loader.js:1548:5)
    at load (node_modules/js-yaml/lib/js-yaml/loader.js:1569:19)
    at Object.safeLoad (node_modules/js-yaml/lib/js-yaml/loader.js:1591:10)
  name: 'YAMLException',
  reason: 'missed comma between flow collection entries',
  mark: 
   Mark {
     name: null,
     buffer: '{\n  "type": "object"\n  "properties": {\n    "example": {\n      "type": "string"\n    }\n  }\n}\n\u0000',
     position: 23,
     line: 2,
     column: 2 },
  message: 'missed comma between flow collection entries at line 3, column 3:\n      "properties": {\n      ^' }
Creating initial clone of data at #/paths/~1example/get/responses/200/content/application~1json/schema
Finished resolution! 
Found 14 rules: parameter-description,operation-operationId,operation-summary-or-description,operation-tags,server-trailing-slash,openapi-tags,openapi-tags-alphabetical,reference-no-other-properties,example-value-or-externalValue,reference-components-regex,no-script-tags-in-markdown,info-contact,license-apimatic-bug,no-eval-in-descriptions
Specification schema is invalid.

#/paths/~1example/get/responses/200/content/application~1json
expected '﻿{\n  "type": "object"\n  "properties": {\n    "example": {\n      "type": "string"\n    }\n  }\n}\n' to be an object
    expected '﻿{\n  "type": "object"\n  "properties": {\n    "example": {\n      "type": "string"\n    }\n  }\n}\n' to have type object
        expected 'string' to be 'object'
```

**Comment by philsturgeon** [comment]
I'm gonna punt this one off to @MikeRalphson I think 😛

**Comment by MikeRalphson** [comment]
I'll make a note to look into whether the target of a `$ref` can ever be other than json or yaml in the upcoming release of `openapi-kit`, but for now I'll expose those warnings even when `verbose` isn't set.

**Comment by philsturgeon** [comment]
This has been solved through improving error messages in #37. Thank you @MikeRalphson!
